include/revng/Model/Importer/Binary/BinaryImporterOptions.h

```diff
@@ -17,14 +17,62 @@ inline llvm::cl::list<std::string> ImportDebugInfo("import-debug-info",
 using OptLevel = llvm::cl::opt<unsigned>;
-inline OptLevel FetchDebugInfoWithLevel("fetch-debuginfo",
-                                        llvm::cl::desc(DESCRIPTION),
-                                        llvm::cl::value_desc("dependency "
-                                                             "level"),
-                                        llvm::cl::cat(MainCategory),
-                                        llvm::cl::init(0));
+inline OptLevel EnableRemoteDebugInfoWithLevel("enable-remote-debug-info",
+                                               llvm::cl::desc(DESCRIPTION),
+                                               llvm::cl::value_desc("dependency"
+                                                                    " "
+                                                                    "level"),
+                                               llvm::cl::cat(MainCategory),
+                                               llvm::cl::init(0));
 #undef DESCRIPTION
 
+enum DebugInfoAction { no, yes, libraries };
+
+#define DESCRIPTION                                                       \
+  "During the import-model phase, this option controls whether we:\n 1) " \
+  "ignore debug info\n 2) consider debug info from the file itself\n 3) " \
+  "consider debug info from libraries that the program depends on\n"
+using DebugInfoOptAction = llvm::cl::opt<DebugInfoAction>;
+inline DebugInfoOptAction
+  DebugInfoAction("debug-info",
+                  llvm::cl::desc(DESCRIPTION),
+                  llvm::cl::value_desc("debug-"
+                                       "info "
+                                       "action"),
+                  llvm::cl::values(clEnumVal(no, "Ignore Debug Info"),
+                                   clEnumVal(yes, "Consider Debug Info"),
+                                   clEnumVal(libraries,
+                                             "Consider Debug Info in "
+                                             "Depending "
+                                             "Libraries")),
+                  llvm::cl::cat(MainCategory),
+                  llvm::cl::init(yes));
+#undef DESCRIPTION
+
+struct DebugInfoOptions {
+  bool IgnoreDebugInfo = false;
+
+  unsigned FetchDebugInfoWithLevel = 0;
+};
+
+namespace {
+inline void parseDebugInfoOptions(DebugInfoOptions &TheDebugInfoOptions) {
+  if (DebugInfoAction == DebugInfoAction::no) {
+    TheDebugInfoOptions.IgnoreDebugInfo = true;
+  } else if (DebugInfoAction == DebugInfoAction::yes) {
+    TheDebugInfoOptions.FetchDebugInfoWithLevel = 1;
+  } else if (DebugInfoAction == DebugInfoAction::libraries) {
+    TheDebugInfoOptions.FetchDebugInfoWithLevel = 2;
+  }
+
+  // If EnableRemoteDebugInfoWithLevel option was set, override the fetching
+  // level.
+  if (EnableRemoteDebugInfoWithLevel != 0)
+    TheDebugInfoOptions
+      .FetchDebugInfoWithLevel = EnableRemoteDebugInfoWithLevel;
+}
```

Comment:
I'm OK with the CLI flags but I'm confused about `DebugInfoOptions`.

I'd like the `struct` to reflect the CLI options, something like:

```c++
struct {
  DebugInfoAction SomeName;
  bool FecthRemote;
}
```

Reply:
cc: @dwarfyresearch

Reply:
@ale, or should I refactor it?

Reply:
Yes please

Reply:
Please review: `BinaryImporter: rework command line interface`.

Reply:
Closing this one in favour of the new threads.